src/core/Entry.cpp

```diff
@@ -218,30 +220,37 @@ QString Entry::defaultAutoTypeSequence() const
     return m_data.defaultAutoTypeSequence;
 }
 
+/**
+ * Determine the effective sequence that will be injected
+ * This function return an empty string if a parent group has autotype disabled or if the entry has no parent
+ */
 QString Entry::effectiveAutoTypeSequence() const
 {
-    if (!m_data.defaultAutoTypeSequence.isEmpty()) {
-        return m_data.defaultAutoTypeSequence;
+    if (autoTypeEnabled() == false) {
+        return QString();
     }
-    QString sequence;
 
-    const Group* grp = group();
-    if(grp) {
-      sequence = grp->effectiveAutoTypeSequence();
-    } else {
-      return QString();
+    const Group* parent = group();
+    if (!parent) {
+        return QString();
     }
 
-    if (sequence.isEmpty() && (!username().isEmpty() || !password().isEmpty())) {
+    QString sequence = parent->effectiveAutoTypeSequence();
+    if (sequence.isEmpty()) {
+        return QString();
+    }
+
+    if (!m_data.defaultAutoTypeSequence.isEmpty()) {
+        return m_data.defaultAutoTypeSequence;
```

Comment:
Why are you returning here with the default sequence? Shouldn't that be the last fallback?

Reply:
The priority is the following:
- Entry has autotype disabled -> empty sequence
- some Group in hierarchy has autotype disabled -> empty sequence
- defaultAutoTypeSequence (means that the entry has a custom sequence set)
- if the entry doesn't have a custom sequence check if some parent Group has one
- if no custom sequence is set all the way to Root, use RootSequence/PasswordOnly/UsernameOnly

it's the same priority list used by KeePass and I've added tests that reflect the above list.

Reply:
Ah, so defaultAutoTypeSequence isn't the default, but the entry-specific sequence? I recommend changing the variable name then.